tawazi/_dag/dag.py

```diff
@@ -417,25 +416,107 @@ def _assign_compound_priority(self) -> None:
             # assign the new leaf nodes
             leaf_ids = graph_ids.leaf_nodes
 
-    def draw(self, k: float = 0.8, display: bool = True, t: int = 3) -> None:
+    def draw(
+        self, show_nodes_names: bool = False, fig_name: str = "", layout_algorithm: str = "kk"
+    ) -> None:
         """Draws the Networkx directed graph.
 
         Args:
-            k (float): parameter for the layout of the graph, the higher, the further the nodes apart. Defaults to 0.8.
-            display (bool): display the layout created. Defaults to True.
-            t (int): time to display in seconds. Defaults to 3.
+            show_nodes_names (bool): whether to display nodes names over datapoints
+            fig_name (str): save in local file path if valid input
+            layout_algorithm (str): layout algorithm for igraph, defaults to kk
         """
-        import matplotlib.pyplot as plt
+        import igraph as ig
+        from plotly.graph_objs import Figure, Layout, Scatter, layout
+
+        nx_graph = self.graph_ids
```

Comment:
no need to declare a variable just for this

Reply:
fixed